protocol/v2/ssv/runner/proposer.go

```diff
@@ -161,6 +138,25 @@ func (r *ProposerRunner) ProcessPreConsensus(logger *zap.Logger, signedMsg *spec
 	return nil
 }
 
+// Function to attempt fetching a beacon block based on the provided conditions.
+// Returns the block (as ssz.Marshaler), the data version, and any error encountered.
+func (r *ProposerRunner) fetchBeaconBlock(logger *zap.Logger, slot phase0.Slot, fullSig []byte) (ssz.Marshaler, spec.DataVersion, error) {
+	var obj ssz.Marshaler
+	var ver spec.DataVersion
+	var err error
+	if r.ProducesBlindedBlocks {
+		// Attempt to fetch the blinded beacon block.
+		obj, ver, err = r.GetBeaconNode().GetBlindedBeaconBlock(slot, r.GetShare().Graffiti, fullSig)
```

Comment:
Did you check how long it usually takes when it fails? Could this duration be an issue? (I think not, just want to be sure)